File: devops/Gestion_Reservation/resource-service/src/main/java/cheikh/bg/resourceservice/service/ResourceService.java
package cheikh.bg.resourceservice.service;

import cheikh.bg.resourceservice.dto.ResourceRequestDTO;
import cheikh.bg.resourceservice.dto.ResourceResponceDTO;

import java.util.List;

public interface ResourceService {
    ResourceResponceDTO addResource(ResourceRequestDTO resourceRequestDTO);
    ResourceResponceDTO getResourceById(Long resourceId);
    List<ResourceResponceDTO> getAllResources();
    ResourceResponceDTO updateResource(Long id,ResourceRequestDTO resourceRequestDTO );

}

File: angular-front-app/src/app/services/resource.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {Resource} from "../../Resource";

@Injectable({
  providedIn: 'root'
})
export class ResourceService {
private apiResourceUrl = 'http://localhost:9999/resource-service/api/resources';

  constructor(private http: HttpClient) {}

  public getResources(): Observable<Resource[]> {
    return this.http.get<Resource[]>(this.apiResourceUrl);
  }

  public addResource(resource: Resource): Observable<Resource> {
    return this.http.post<Resource>(`${this.apiResourceUrl}/create`, resource);
  }

  public updateResource(resourceId: number, resource: Resource): Observable<Resource> {
    return this.http.put<Resource>(`${this.apiResourceUrl}/update/${resourceId}`, resource);
  }
  public getResourceById(resourceId: number): Observable<Resource> {
    return this.http.get<Resource>(`${this.apiResourceUrl}/${resourceId}`);
  }

  public deleteResource(personId: number): Observable<void> {
    return this.http.delete<void>(`${this.apiResourceUrl}/delete/${personId}`);
  }
}
